diff --git a/node_modules/@mccsoft/react-native-matomo/android/build.gradle b/node_modules/@mccsoft/react-native-matomo/android/build.gradle
index 81c51a4..4902ac0 100644
--- a/node_modules/@mccsoft/react-native-matomo/android/build.gradle
+++ b/node_modules/@mccsoft/react-native-matomo/android/build.gradle
@@ -144,7 +144,7 @@ repositories {
 dependencies {
     //noinspection GradleDynamicVersion
   implementation "com.facebook.react:react-native:${safeExtGet('reactNativeVersion', '+')}"
-  implementation 'com.github.matomo-org:matomo-sdk-android:4.1.4'
+  implementation 'com.github.matomo-org:matomo-sdk-android:4.2'
 // From node_modules
 }
 
diff --git a/node_modules/@mccsoft/react-native-matomo/mccsoft-react-native-matomo.podspec b/node_modules/@mccsoft/react-native-matomo/mccsoft-react-native-matomo.podspec
index d85fb73..ca0a623 100644
--- a/node_modules/@mccsoft/react-native-matomo/mccsoft-react-native-matomo.podspec
+++ b/node_modules/@mccsoft/react-native-matomo/mccsoft-react-native-matomo.podspec
@@ -12,14 +12,14 @@ Pod::Spec.new do |s|
   s.authors       = package["author"]
 
   s.swift_version = '5.0'
-  s.platforms     = { :ios => "10.0" }
+  s.platforms     = { :ios => "11.0" }
   s.source        = { :git => "https://github.com/mccsoft/react-native-matomo.git", :tag => "#{s.version}" }
 
 
   s.source_files  = "ios/**/*.{h,m,mm,swift}"
 
   s.dependency "React-Core"
-  s.dependency "MatomoTracker"
+  s.dependency "MatomoTracker", "7.6.0"
 
   # Don't install the dependencies when we run `pod install` in the old architecture.
   if ENV['RCT_NEW_ARCH_ENABLED'] == '1' then
